near/src/runtime.rs

```diff
@@ -793,6 +794,27 @@ impl RuntimeAdapter for NightshadeRuntime {
         epoch_manager.get_epoch_start_height(block_hash).map_err(Error::from)
     }
 
+    fn get_gc_stop_height(&self, block_hash: &CryptoHash) -> Result<BlockHeight, Error> {
+        let mut epoch_manager = self.epoch_manager.write().expect(POISONED_LOCK_ERR);
+        // an epoch must have a first block.
+        let epoch_first_block = epoch_manager.get_block_info(block_hash)?.epoch_first_block;
+        let epoch_first_block_info = epoch_manager.get_block_info(&epoch_first_block)?;
+        // maintain pointers to avoid cloning.
+        let mut last_block_in_prev_epoch = epoch_first_block_info.prev_hash;
+        let mut epoch_start_height = epoch_first_block_info.height;
+        for _ in 0..NUM_EPOCHS_TO_KEEP_STORE_DATA - 1 {
+            let epoch_first_block = match epoch_manager.get_block_info(&last_block_in_prev_epoch) {
+                Ok(block_info) => block_info.epoch_first_block,
+                Err(EpochError::MissingBlock(_)) => return Ok(epoch_start_height),
```

Comment:
shall this return the tail instead?

Reply:
What do you mean by tail?

Reply:
It's the same things I think. Every epoch must have a last block so the only way this can happen is that we are at genesis block, which means this would return genesis block height.